micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -295,7 +295,7 @@ void registerQueryMetric(Statistics statistics) {
             for (String query : statistics.getQueries()) {
                 QueryStatistics queryStatistics = statistics.getQueryStatistics(query);
                 if (Objects.nonNull(queryStatistics)) {
-                    String queryName = query.replace(" ", "_").toLowerCase();
+                    String queryName = query.replace(" ", "_").replace(".", "_").toLowerCase();
```

Comment:
Dots are OK in tags, so they shouldn't cause a problem (at least in Prometheus).

Reply:
Graphite is treating dots as next tag

Reply:
Could you create an issue about the graphite registry not escaping dots in tags?

Reply:
Sure, will create

Reply:
If the intent is to generically make the query fit the convention of the MeterRegistry, the NamingConvention configured for the MeterRegistry can be used: `meterRegistry.config().namingConvention().tagValue(query)`

Reply:
Yup, that was my intention, since graphite problem is resolved we can proceed with this generic approach